readthedocs/proxito/views/serve.py

```diff
@@ -45,9 +45,16 @@ def get(self,
     ):  # noqa
         """Take the incoming parsed URL's and figure out what file to serve."""
 
+        # Handle external domain
+        if request.external_domain:
+            self.version_type = EXTERNAL
+            log.warning('Replacing version slug from URL old=%s new=%s',
+                        version_slug, request.host_version_slug)
+            version_slug = request.host_version_slug
```

Comment:
This logic should go into some kind of helper, just haven't set it up yet.

Reply:
Yes. This could probably be a method of `ServeDocsMixin`.